pkg/repo/etcd.go

```diff
@@ -11,18 +11,27 @@ import (
 
 type Etcd interface {
 	Put(ctx context.Context, cli *clientv3.Client, key string, value string) error
-	Client(cfg EtcdConfig) (*clientv3.Client, error)
+
+	Client() (*clientv3.Client, error)
+
+	Watch(ctx context.Context, cli *clientv3.Client, key string) (clientv3.WatchChan, context.CancelFunc)
 }
```

Comment:
should we tried to hide etcd client library behind this interface?
if rest of the code still had to be aware of etcd client library, what benefit do we get from this abstraction?

Reply:
I found writing tests easier how repos that use etcd.

Reply:
yes but if you fully hide `clientv3`, it'll be even more simplier, probably you'll need just few methods like Put, Get, Watch, all of them can accept just base types like strings, for watch you have 2 options ether to return a channel or accept call back for processing update.

Reply:
Yes but I want to mock client creation too `clientv3.New`. Because if we do not mock it, the client creation method tried to initiate grpc connection and create real instances of watcher etc..